alg/teca_2d_component_area.cxx

```diff
@@ -285,6 +288,20 @@ const_p_teca_dataset teca_2d_component_area::execute(
 
     teca_metadata &out_metadata = out_mesh->get_metadata();
 
+    // get the background_id, and pass it through
+    long bg_id = this->background_id;
+    if (this->background_id < 0)
+    {
+        if (in_metadata.get("background_id", bg_id))
+        {
+            TECA_ERROR("Metadata is missing the key \"background_id\". "
+                "One should specify it via the \"background_id\" algorithm "
+                "property")
+            return nullptr;
+        }
+    }
+    out_metadata.set("background_id", bg_id);
```

Comment:
Should we set `this->background_id = bg_id` after line 303?

Also, it looks like the background id isn't considered in `component_area`; if someone were to pass in a labeled array, where the background is labeled -1 or some other out-of-bounds value, then I think that line 97 would cause a segmentation fault.

I think this could be fixed by changing line 97 to something like
```
// set the index of the current component
int area_index = labels[jj + i];
// if it is the background component, make sure the index is 0
if (area_index == this->background_id) area_index -= this->background_id;
area[labels[jj + i]] += rho_sq_d_theta[i]*d_cos_phi_j;
```

Reply:
> Should we set this->background_id = bg_id after line 303?

I don't think so. We don't want to modify the state of the class itself. The only way  `this->background_id` changes is if the user sets it at runtime.

>  the background id isn't considered in component_area

I don't think it needs, but the background id does need to be passed through. & it is.

> if someone were to pass in a labeled array, where the background is labeled -1 or some other out-of-bounds value

In that case the function you reference, the one optimized for contiguous labels starting at 0, can't be used. Instead the implementation that does an indirect lookup must be used. This is controlled by the `contiguous_component_ids` algorithm property.

Reply:
got it -- I hadn't caught that an associative array would be passed in; makes sense.  Thanks.

Reply:
sorry, one more thought: should there be a check for consistency between `contiguous_component_ids` and `background_id`?  I someone sets the background to a negative value and sets `contiguous_component_ids`, then I think that could lead to a segfault; might be better to error out if so

Reply:
True. and we could scan the label values to see if they are indeed consecutive and start at 0 rather than have a flag that the user would set. I'll make this a new issue though

Reply:
see issue #410